### ISSUE TITLE
UI crashes when entering to the one of the project submodules

### ISSUE THREAD

**Issue by vikulikov** [extrawurst/gitui#1510]
**Describe the bug**
Pressing `shift + s` window with submodules opens. After selecting one of the submodules with `enter` and waiting for 2-3 seconds UI of `gitui` closes but the app is seemed to be still working because shell is unresponsive.
I have uncommited changes in the submodules. Maybe the problem is around it.

**To Reproduce**
1. Open gitui in the project with submodules
2. Press `Shift + s` to open submodules view
3. Select any submodule with `enter`
4. Wait for 2-3 secs
5. See error

**Expected behavior**
To not crash

**Context:**
 - `macOS 13.1`
 - `gitui 0.22.1`
 - terminal `alacritty 0.11.0 (8dbaa0b)`
 - `rustc 1.66.0 (69f9c33d7 2022-12-12)`

**Comment by vikulikov** [comment]
Tell me please how I can give you additional context. Because I don't see any errors on crash

**Comment by extrawurst** [comment]
can you run with logging enabled and share that?

**Comment by vikulikov** [comment]
[gitui.log](https://github.com/extrawurst/gitui/files/10425577/gitui.log)

**Comment by suricactus** [comment]
Same behaviour here, Ubuntu 22.04. Stable reproduction:

1) Install latest gitui (0.22.1)
```
cargo install gitui
```
2) Checkout repo
```
git clone https://github.com/opengisch/qfieldsync
cd qfieldsync
```
3) Run `gitui` in the repo dir
4) Press `shift+s`
5) Select submodule `libqfieldsync`
6) Wait 1-2 seconds
7) GUI crashes

Logs:
```
13:16:03 [TRACE] (6) mio::poll: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.8.5/src/poll.rs:532] registering event source with poller: token=Token(0), interests=READABLE
13:16:03 [TRACE] (6) mio::poll: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.8.5/src/poll.rs:532] registering event source with poller: token=Token(1), interests=READABLE
13:16:03 [TRACE] (1) gitui::watcher: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/watcher.rs:21] poll watcher: false recommended: Inotify
13:16:03 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:119] open repo at: RefCell { value: Path(".") }
13:16:03 [TRACE] (7) mio::poll: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.8.5/src/poll.rs:532] registering event source with poller: token=Token(0), interests=READABLE
13:16:03 [TRACE] (1) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:102] request: [hash: 4175380985501547786] (type: WorkingDir)
13:16:03 [TRACE] (1) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:102] request: [hash: 5640907937309220126] (type: Stage)
13:16:03 [TRACE] (5) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:160] status fetched: 5640907937309220126 (type: Stage)
13:16:03 [TRACE] (1) gitui: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/main.rs:200] app start: 20 ms
13:16:03 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:477] update
13:16:03 [TRACE] (1) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:96] request blocked, still pending
13:16:03 [TRACE] (1) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:102] request: [hash: 9398681960897875946] (type: Stage)
13:16:03 [TRACE] (2) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:160] status fetched: 9398681960897875946 (type: Stage)
13:16:03 [TRACE] (3) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:160] status fetched: 4175380985501547786 (type: WorkingDir)
13:16:03 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:496] update_async: Git(Status)
13:16:03 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:496] update_async: Git(Branches)
13:16:03 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:496] update_async: Git(Status)
13:16:03 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:496] update_async: Git(Status)
13:16:05 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:382] event: Input(Key(KeyEvent { code: Char('S'), modifiers: SHIFT, kind: Press, state: NONE }))
13:16:05 [TRACE] (1) asyncgit::sync::submodules: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/sync/submodules.rs:118] [sub] repo_wd: "/home/suricactus/work/opengis/qfieldsync/"
13:16:05 [TRACE] (1) asyncgit::sync::submodules: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/sync/submodules.rs:119] [sub] repo_path: "/home/suricactus/work/opengis/qfieldsync/.git/"
13:16:05 [TRACE] (1) asyncgit::sync::submodules: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/sync/submodules.rs:122] [sub] parent_path: "/home/suricactus/work/opengis"
13:16:07 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:382] event: Input(Key(KeyEvent { code: Enter, modifiers: NONE, kind: Press, state: NONE }))
13:16:07 [TRACE] (1) gitui::watcher: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/watcher.rs:21] poll watcher: false recommended: Inotify
13:16:07 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:119] open repo at: RefCell { value: Path("/home/suricactus/work/opengis/qfieldsync/qfieldsync/libqfieldsync") }
13:16:07 [TRACE] (11) mio::poll: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/mio-0.8.5/src/poll.rs:532] registering event source with poller: token=Token(0), interests=READABLE
13:16:07 [TRACE] (1) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:102] request: [hash: 3339512511350449500] (type: WorkingDir)
13:16:07 [TRACE] (1) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:102] request: [hash: 5664221600382528033] (type: Stage)
13:16:07 [TRACE] (1) gitui: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/main.rs:200] app start: 3254 ms
13:16:07 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:477] update
13:16:07 [TRACE] (1) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:96] request blocked, still pending
13:16:07 [TRACE] (1) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:96] request blocked, still pending
13:16:07 [TRACE] (4) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:160] status fetched: 5664221600382528033 (type: Stage)
13:16:07 [TRACE] (3) asyncgit::status: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/asyncgit-0.22.1/src/status.rs:160] status fetched: 3339512511350449500 (type: WorkingDir)
13:16:07 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:496] update_async: Git(Branches)
13:16:07 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:496] update_async: Git(Status)
13:16:07 [TRACE] (1) gitui::app: [/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/app.rs:496] update_async: Git(Status)
13:16:09 [DEBUG] (8) gitui::watcher: notify events: 1
13:16:09 [DEBUG] (8) gitui::watcher: notify [0]: DebouncedEvent { path: "/home/suricactus/work/opengis/qfieldsync/qfieldsync/libqfieldsync/.git/gitui", kind: Any }
13:16:09 [ERROR] panic: PanicInfo { payload: Any { .. }, message: Some(send error: "SendError(..)"), location: Location { file: "/home/suricactus/.cargo/registry/src/github.com-1ecc6299db9ec823/gitui-0.22.1/src/watcher.rs", line: 69, col: 37 }, can_unwind: true }
trace:
   0: gitui::set_panic_handlers::{{closure}}
   1: <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/alloc/src/boxed.rs:2032:9
      std::panicking::rust_panic_with_hook
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:692:13
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:579:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys_common/backtrace.rs:137:18
   4: rust_begin_unwind
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/panicking.rs:575:5
   5: core::panicking::panic_fmt
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/panicking.rs:64:14
   6: core::result::unwrap_failed
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/core/src/result.rs:1791:5
   7: std::sys_common::backtrace::__rust_begin_short_backtrace
   8: core::ops::function::FnOnce::call_once{{vtable.shim}}
   9: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/alloc/src/boxed.rs:2000:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/alloc/src/boxed.rs:2000:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/fc594f15669680fa70d255faec3ca3fb507c3405/library/std/src/sys/unix/thread.rs:108:17
  10: start_thread
             at ./nptl/pthread_create.c:442:8
  11: clone3
             at ./misc/../sysdeps/unix/sysv/linux/x86_64/clone3.S:81
```

**Comment by extrawurst** [comment]
closed via 94424eb
